app/models/bench.rb

```diff
@@ -23,6 +28,50 @@ class Bench < ApplicationRecord
            class_name: 'RequestDistribution',
            inverse_of: :bench,
            dependent: :restrict_with_error
+  # Checks
+  def dimensions_must_be_strictly_positive
+    return unless dimensions
+
+    return unless dimensions.any? { |d| d <= 0 }
+
+    errors.add(:dimensions, 'each dimension must be greater than 0')
+  end
+
+  def positions_must_be_positive
+    return unless positions
+
+    return unless positions.any?(&:negative?)
+
+    errors.add(:positions, 'each position must be positive')
+  end
+
+  def overlapping_bench_exists
+    return if errors[:dimensions].any? || errors[:positions].any?
+    return unless greenhouse
+
+    new_bench_position = positions
+    new_bench_dimensions = dimensions
+
+    if greenhouse.benches.any? do |bench|
+      next if bench == self
+
+      existing_bench_position = bench.positions
+      existing_bench_dimensions = bench.dimensions
+
+      positions_overlap?(new_bench_position, new_bench_dimensions, existing_bench_position, existing_bench_dimensions)
+    end
+      errors.add(:positions, 'bench overlaps with an existing bench')
+    end
+  end
+
+  def positions_overlap?(pos1, dim1, pos2, dim2)
+    x1, y1 = pos1
+    width1, height1 = dim1
+    x2, y2 = pos2
+    width2, height2 = dim2
+
+    x1 < x2 + width2 && x1 + width1 > x2 && y1 < y2 + height2 && y1 + height1 > y2
+  end
```

Comment:
cette méthode n'est pas très "orientée objet" car elle n'utilise aucun argument du bench courant.
Elle pourrait simplement comparer notre bench actuel avec un autre bench donné en entrée, par exemple comme ça :
```suggestion
  def positions_overlap?(other_bench)
    x1, y1 = positions
    width1, height1 = dimensions
    x2, y2 = other_bench.positions
    width2, height2 = other_bench.dimensions

    x1 < x2 + width2 && x1 + width1 > x2 && y1 < y2 + height2 && y1 + height1 > y2
  end
```
ça a un peu plus de sens, car comme ça cette méthode propre à un bench (car elle est définie sur le modèle) utilise bien les données du dit bench :) 

et au-dessus, tu peux l'appeler en faisant simplement `positions_overlap?(bench)`, et ainsi, t'affranchir de la définition des variables `new_bench_position`, `new_bench_dimensions`, `existing_bench_position` et `existing_bench_dimensions` ! :) 
Ça allégera pas mal le code.